server/app/controllers/lists.js

```diff
@@ -0,0 +1,134 @@
+const ErrorResponse = require('../utils/errorResponse');
+const asyncHandler = require('../middleware/async');
+const List = require('../models/List');
+
+/**
+ * @api {get} api/v1/users/:userId/lists
+ * @apiName getLists
+ * @apiGroup lists
+ *
+ * @apiDescription Get all of the lists for a user
+ */
+// eslint-disable-next-line no-unused-vars
+const getLists = asyncHandler(async (req, res, next) => {
+  const { userId } = req.params;
+  const lists = await List.find({ user: userId });
+  if (!lists) {
+    return next(
+      new ErrorResponse(
+        `No lists with the user id of ${req.params.userId}`,
+        404,
+      ),
+    );
+  }
+  return res.status(200).json({ success: true, lists });
+});
+
+/**
+ * @api {get} api/v1/users/:userId/lists/:listId
+ * @apiName getList
+ * @apiGroup lists
+ *
+ * @apiDescription Get one list for the user
+ */
+// eslint-disable-next-line no-unused-vars
+const getList = asyncHandler(async (req, res, next) => {
+  const { listId } = req.params;
+  const lists = await List.findById(listId);
+  if (!lists) {
+    return next(
+      new ErrorResponse(`No lists with the id of ${req.params.listId}`, 404),
+    );
+  }
+  return res.status(200).json({ success: true, lists });
+});
+
+/**
+ * @api {post} /api/v1/lists
+ * @apiName postList
+ * @apiGroup lists
+ * @apiPermission protected
+ *
+ * @apiDescription Creates user in database
+ */
+// eslint-disable-next-line no-unused-vars
+const postList = asyncHandler(async (req, res, next) => {
+  const { name, coverUrl, user, products } = req.body;
+
+  // Create list
+  const list = await List.create({
+    name,
+    coverUrl,
+    user,
+    products,
+  });
+
+  res.status(200).json({
+    success: true,
+    list: list.toJSON(),
+  });
+});
+
+/**
+ * @api {update} /api/v1/lists/:listId
+ * @apiName updateList
+ * @apiGroup lists
+ * @apiPermission protected
+ *
+ * @apiDescription Update a single list
+ */
+// eslint-disable-next-line no-unused-vars
+const updateList = asyncHandler(async (req, res, next) => {
+  const { name, coverUrl, user, products, id } = req.body;
+
+  // Create list
+  const list = await List.findByIdAndUpdate(id, {
+    name,
+    coverUrl,
+    user,
+    ...products,
+  });
+
+  res.status(200).json({
+    success: true,
+    list: list.toJSON(),
+  });
+});
```

Comment:
After reviewing the sketch, I don't think we need this route since the only time the list is updated is when a new product is added to the list, in which case a `POST` to `/api/v1/lists/:listId/products` is made instead.